autoload/neoterm/test/golang.vim

```diff
@@ -0,0 +1,22 @@
+function! neoterm#test#golang#run(scope)
+  let command = 'go test'
+
+  if a:scope == 'file'
```

Comment:
I know nothing about go-lang, but I guess you can just pass the file path to `go test <file_path>` and it should work. I saw [this documentation](http://golang.org/doc/code.html#Testing). 

You can see how to do this in `viml` looking the [minitest lib](https://github.com/kassio/neoterm/blob/master/autoload/neoterm/test/minitest.vim#L5).